OpenSSL/SSL.py

```diff
@@ -310,6 +310,12 @@ def load_verify_locations(self, cafile, capath=None):
         :param capath: In which directory we can find the certificates
         :return: None
         """
+
+        # Backward compatibility
+        if isinstance(cafile, _text_type):
+            DeprecationWarning("str object in cafile is no longer accepted, use bytes")
```

Comment:
This doesn't actually emit a warning.  It just creates the warning object and then discards it.  Warnings are typically emitted using `warnings.warn`.

Also, this message might be confusing on Python 2 where `str` and `bytes` are synonyms.  I suggest saying "text" instead of "str object" or, perhaps better, emit different messages on Python 2 and Python 3: say "unicode" on Python 2 and "str" on Python 3.